### DISCUSSION TITLE
DataTable - listening to end of sorting

### DISCUSSION THREAD

**Discussion by hgfrey** [primefaces/community#515] [PrimeFaces] [answered]
I need information when the end of the sorting has been reached because I need the newly sorted data rows. The Litener is executed even before the sorted list is set in the backend and oncomplete is not executed when the sort icon in the ui is clicked. How can I know that the sorted list has now arrived in the backend?

Thank you and kind regards

**Comment by tandraschko** [comment, marked as answer]
`<f:event type="org.primefaces.event.data.PostSortEvent" listener="..."/>`
should do the trick

**Reply by hgfrey** [reply]
Thank you! This works for me.
